File: client/clientmanager/src/main/java/io/mosip/registration/clientmanager/dto/registration/BiometricsDto.java
package io.mosip.registration.clientmanager.dto.registration;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class BiometricsDto {

    private String modality;
    private String bioSubType;
    private String bioValue;
    private String specVersion;
    private boolean isException;
    private String decodedBioResponse;
    private String signature;
    private boolean isForceCaptured;
    private int numOfRetries;
    private double sdkScore;
    private float qualityScore;
}

File: lib/model/biometrics_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'biometrics_dto.freezed.dart';

part 'biometrics_dto.g.dart';

@freezed
class BiometricsDto with _$BiometricsDto {
  const factory BiometricsDto({
    String? modality,
    String? bioSubType,
    String? bioValue,
    String? specVersion,
    bool? isException,
    String? decodedBioResponse,
    String? signature,
    bool? isForceCaptured,
    int? numOfRetries,
    double? sdkScore,
    double? qualityScore,
  }) = _BiometricsDto;

  factory BiometricsDto.fromJson(Map<String, Object?> json) =>
      _$BiometricsDtoFromJson(json);
}
